### PR TITLE
Feature/gwt 50

### PR DESCRIPTION
**Pull request by NickNish09**
https://github.com/RibonDAO/interface/assets/27820474/aeb4604f-4761-4d0a-85cb-0acb56047c57

# Description

- Add email input section
Refactor donation flow when the user is not authenticated

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test A

### CODE REVIEW

File: src/pages/donations/ConfirmDonationPage/SignedInSection/index.tsx [hunk 1]
```diff
@@ -18,6 +19,10 @@ function SignedInSection({ nonProfit, onContinue }: Props): JSX.Element {
   const { formattedImpactText } = useFormattedImpactText();
   const { currentUser } = useCurrentUser();
 
+  useEffect(() => {
+    newLogEvent("view", "P1_donateConfirmModal");
```
Comment (thaisribon) [inline src/pages/donations/ConfirmDonationPage/SignedInSection/index.tsx:23]: this name doesn't make sense anymore :/ do you think we should keep it this way for a while?

Comment (NickNish09) [reply src/pages/donations/ConfirmDonationPage/SignedInSection/index.tsx:23]: I think there will be tasks to review the flow. Meanwhile I was keeping the same events so it doesnt affect analytics

File: src/pages/donations/ConfirmDonationPage/EmailInputSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import BackgroundShapes from "assets/images/background-shapes.svg";
+import { isValidEmail } from "lib/validators";
+import { newLogEvent } from "lib/events";
+import * as S from "./styles";
+
+type Props = {
+  nonProfit: NonProfit;
+  onContinue: (email: string) => void;
+};
+function EmailInputSection({ nonProfit, onContinue }: Props): JSX.Element {
```
[the remark this answers is not in the document]
Comment (NickNish09) [reply src/pages/donations/ConfirmDonationPage/EmailInputSection/index.tsx]: good call, changed

File: public/locales/en/translation.json [hunk 3]
```diff
@@ -118,6 +118,16 @@
       "signedInSection": {
         "title": "Your donation",
         "confirmText": "Confirm donation"
+      },
+      "emailInputSection": {
```
Comment (thaisribon) [inline public/locales/en/translation.json:122]: signInSection?

Comment (NickNish09) [reply public/locales/en/translation.json]: fixed

**Review by thaisribon — APPROVED** [review]

**Pull request merged**